File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Slb/AlibabaCloud/SDK/ROS/CDK/Slb/Datasource/IDomainExtension.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Slb.Datasource
{
    /// <summary>Represents a `DomainExtension`.</summary>
    [JsiiInterface(nativeType: typeof(IDomainExtension), fullyQualifiedName: "@alicloud/ros-cdk-slb.datasource.IDomainExtension")]
    public interface IDomainExtension : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute Domain: The domain name.</summary>
        [JsiiProperty(name: "attrDomain", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrDomain
        {
            get;
        }

        /// <summary>Attribute DomainExtensionId: The ID of the additional certificate.</summary>
        [JsiiProperty(name: "attrDomainExtensionId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrDomainExtensionId
        {
            get;
        }

        /// <summary>Attribute ListenerPort: The frontend port of the HTTPS listener that is configured for the SLB instance.</summary>
        /// <remarks>
        /// Valid values: 1 to 65535.
        /// </remarks>
        [JsiiProperty(name: "attrListenerPort", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrListenerPort
        {
            get;
        }

        /// <summary>Attribute LoadBalancerId: The ID of the SLB instance.</summary>
        [JsiiProperty(name: "attrLoadBalancerId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrLoadBalancerId
        {
            get;
        }

        /// <summary>Attribute ServerCertificateId: The ID of the server certificate that is used by the domain name.</summary>
        [JsiiProperty(name: "attrServerCertificateId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrServerCertificateId
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-slb.datasource.DomainExtensionProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Slb.Datasource.IDomainExtensionProps Props
        {
            get;
        }

        /// <summary>Represents a `DomainExtension`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IDomainExtension), fullyQualifiedName: "@alicloud/ros-cdk-slb.datasource.IDomainExtension")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Slb.Datasource.IDomainExtension
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute Domain: The domain name.</summary>
            [JsiiProperty(name: "attrDomain", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrDomain
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute DomainExtensionId: The ID of the additional certificate.</summary>
            [JsiiProperty(name: "attrDomainExtensionId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrDomainExtensionId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute ListenerPort: The frontend port of the HTTPS listener that is configured for the SLB instance.</summary>
            /// <remarks>
            /// Valid values: 1 to 65535.
            /// </remarks>
            [JsiiProperty(name: "attrListenerPort", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrListenerPort
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute LoadBalancerId: The ID of the SLB instance.</summary>
            [JsiiProperty(name: "attrLoadBalancerId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrLoadBalancerId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute ServerCertificateId: The ID of the server certificate that is used by the domain name.</summary>
            [JsiiProperty(name: "attrServerCertificateId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrServerCertificateId
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-slb.datasource.DomainExtensionProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Slb.Datasource.IDomainExtensionProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Slb.Datasource.IDomainExtensionProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkslb/datasource/IDomainExtension.go
package datasource

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkslb/datasource/internal"
)

// Represents a `DomainExtension`.
type IDomainExtension interface {
	alicloudroscdkcore.IResource
	// Attribute Domain: The domain name.
	AttrDomain() interface{}
	// Attribute DomainExtensionId: The ID of the additional certificate.
	AttrDomainExtensionId() interface{}
	// Attribute ListenerPort: The frontend port of the HTTPS listener that is configured for the SLB instance.
	//
	// Valid values: 1 to 65535.
	AttrListenerPort() interface{}
	// Attribute LoadBalancerId: The ID of the SLB instance.
	AttrLoadBalancerId() interface{}
	// Attribute ServerCertificateId: The ID of the server certificate that is used by the domain name.
	AttrServerCertificateId() interface{}
	Props() *DomainExtensionProps
}

// The jsii proxy for IDomainExtension
type jsiiProxy_IDomainExtension struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IDomainExtension) AttrDomain() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrDomain",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IDomainExtension) AttrDomainExtensionId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrDomainExtensionId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IDomainExtension) AttrListenerPort() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrListenerPort",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IDomainExtension) AttrLoadBalancerId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrLoadBalancerId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IDomainExtension) AttrServerCertificateId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrServerCertificateId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IDomainExtension) Props() *DomainExtensionProps {
	var returns *DomainExtensionProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

